helpers/foundation-deployer/stages/apply.go

```diff
@@ -0,0 +1,284 @@
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/mitchellh/go-testing-interface"
+
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/gcp"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/msg"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/steps"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/utils"
+)
+
+
+func DeployBootstrapStage(t testing.TB, s steps.Steps, tfvars GlobalTFVars, c CommonConf) error {
+	repo := "gcp-bootstrap"
+	step := "0-bootstrap"
+
+	bootstrapTfvars := BootstrapTfvars{
+		OrgID:              tfvars.OrgID,
+		DefaultRegion:      tfvars.DefaultRegion,
+		BillingAccount:     tfvars.BillingAccount,
+		GroupOrgAdmins:     tfvars.GroupOrgAdmins,
+		GroupBillingAdmins: tfvars.GroupBillingAdmins,
+		ParentFolder:       tfvars.ParentFolder,
+		ProjectPrefix:      tfvars.ProjectPrefix,
+		FolderPrefix:       tfvars.FolderPrefix,
+		BucketForceDestroy: tfvars.BucketForceDestroy,
+	}
+
+	err := utils.WriteTfvars(filepath.Join(c.FoundationPath, step, "terraform.tfvars"), bootstrapTfvars)
+	if err != nil {
+		return err
+	}
+
+	terraformDir := filepath.Join(c.FoundationPath, step)
+	options := &terraform.Options{
+		TerraformDir: terraformDir,
+		Logger:       c.Logger,
+		NoColor:      true,
+	}
+	// terraform deploy
+	_, err = terraform.InitE(t, options)
+	if err != nil {
+		return err
+	}
+	_, err = terraform.PlanE(t, options)
+	if err != nil {
+		return err
+	}
+
+	// Runs gcloud terraform vet on the
+	if tfvars.HasValidatorProj() {
+		fmt.Println("")
+		fmt.Println("# Running gcloud terraform vet for validation of bootstrap stage")
+		fmt.Println("")
+		err = TerraformVet(t, terraformDir, filepath.Join(c.FoundationPath, "policy-library"), *tfvars.ValidatorProjectId, c)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = terraform.ApplyE(t, options)
+	if err != nil {
+		return err
+	}
+
+	// read bootstrap outputs
+	defaultRegion := terraform.OutputMap(t, options, "common_config")["default_region"]
+	cbProjectID := terraform.Output(t, options, "cloudbuild_project_id")
+	backendBucket := terraform.Output(t, options, "gcs_bucket_tfstate")
+	backendBucketProjects := terraform.Output(t, options, "projects_gcs_bucket_tfstate")
+
+	// replace backend and terraform init migrate
+	err = s.RunStep("gcp-bootstrap.migrate-state", func() error {
+		options.MigrateState = true
+		err = utils.CopyFile(filepath.Join(options.TerraformDir, "backend.tf.example"), filepath.Join(options.TerraformDir, "backend.tf"))
+		if err != nil {
+			return err
+		}
+		err = utils.ReplaceStringInFile(filepath.Join(options.TerraformDir, "backend.tf"), "UPDATE_ME", backendBucket)
+		if err != nil {
+			return err
+		}
+		_, err := terraform.InitE(t, options)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
```suggestion
		return err
```

Reply:
fixed here and in all other cases thru the code